### ISSUE TITLE
Prediction for list-type questions.

### ISSUE THREAD

**Issue by ardakdemir** [dmis-lab/bioasq-biobert#4]
Dear @wonjininfo,

May I kindly ask you to give more details about the inference process for list-type questions. I could not find it in the relevant paper. As far as I have understood, by going over the source-code, the model uses the start and end logits to find the N-highest scoring start+end scores, without conditioning the scores to each other (i.e. end score for a given word when the start word is j and vice versa).

Thanks in advance

Arda

**Comment by wonjininfo** [comment]
Hi Arda, 
Many apologies for my belated reply. I was busy preparing a paper for a conference.

Regarding your question, the main idea of our model for the list questions are almost same with factoid question, which is identical with BERT-QA for SQuAD dataset. 
However, we consider the probability of output (which is start+end score as you mentioned) and threshold them for answering multiple outputs. 

I regret to tell you that I could not understand your mention "without conditioning the scores to each other". Would you please elaborate the phrase?

Thank you for your interest in our paper!
Best, 
Wonjin

**Comment by ardakdemir** [comment]
Wiese et al (2017) uses the idea I mentioned in their question answering model, so I thought it would be easiest to cite the relevant part of their paper here:

<img width="790" alt="Screen Shot 2020-02-12 at 18 12 00" src="https://user-images.githubusercontent.com/12223406/74320037-4a602d80-4dc3-11ea-94bc-624569776e69.png">

The idea is to calculate the likelihood to have an end-word at position j given that the span starts from word i. I hope this explains better what I mentioned previously.

Sorry for not explaining clearly enough.

[Wiese et al]. Neural domain adaptation for biomedical question
answering. In: Proceedings of the 21st Conference on Computational
Natural Language Learning (CoNLL 2017), Vancouver, Canada. pp.
281–289. Association for Computational Linguistics. https://www.aclweb.
org/anthology/K17-1029.

**Comment by wonjininfo** [comment]
Hi Arda, 
Yes, we did not apply the mentioned idea. We only restricted the order of the start and end position tokens . 
I guess it's here :
```python
          if end_index < start_index:
            continue
```
Thanks!

**Comment by ardakdemir** [comment]
Thanks a lot for the clarification!
